src/net/libwebsocketsIO.cpp

```diff
@@ -355,6 +361,11 @@ int LibwebsocketsClient::wsCallback(struct lws *wsi, enum lws_callback_reasons r
             {
                 X509_STORE_CTX_set_error((X509_STORE_CTX*)user, X509_V_OK);
             }
+            else
+            {
+                X509_STORE_CTX_set_error((X509_STORE_CTX*)user, X509_V_ERR_UNSPECIFIED);
+                return 1;
```

Comment:
Usually this method returns `-1` upon error, and `0` when the reason has no specific handler. Since it's a critical zone, I'd double-check that, by returning `1`, libwebsocket will not reuse the failed cert in any way, but reset the SSL negotiation from the scratch.
A possible way to check is that, once this piece of code is executed, the connection is finally established to the right IP address (aka. chatd/presenced).